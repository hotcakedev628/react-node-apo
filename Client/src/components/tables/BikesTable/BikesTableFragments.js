import { NavLink } from 'react-router-dom';
import { s3Url } from '../../../config/constants';
import React from 'react'

export const BasicTableHeaders = () => (
    <React.Fragment>
        <th>Image</th>
        <th>Model</th>
        <th>Color</th>
        <th>Weight</th>
    </React.Fragment>
)

export const AdminTableHeaders = () => (
    <React.Fragment>
        <th className="text-center">Update</th>
        <th className="text-center">Delete</th>
    </React.Fragment>
)

export const BasicTableDescription = ({ item }) => (
    <React.Fragment>
        <td>
            {item.imageName ?
                <img style={{ maxHeight: "60px" }} alt="bike" src={s3Url + item.imageName}></img>
                :
                <p>No image available</p>
            }
        </td>
        <td>{item.model}</td>
        <td>{item.color}</td>
        <td>{item.weight}</td>
    </React.Fragment>
)



export const AdminTableDescriptionActions = ({ item, onDeleteClick }) => (
    <React.Fragment>
        <td className="text-center">
            <NavLink to={'/editBike/' + item._id} className="fa fa-edit text-warning" id="editButton"></NavLink>
        </td>
        <td className="text-center ">
            <a className="link" onClick={() => onDeleteClick(item)}>
                <i className="fa fa-trash text-danger"> </i>
            </a>
        </td>
    </React.Fragment>
)

