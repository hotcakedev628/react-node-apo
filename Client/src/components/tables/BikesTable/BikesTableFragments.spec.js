import React from 'react';
import renderer from 'react-test-renderer';
import { BasicTableDescription } from './BikesTableFragments';



describe('BasicTableDescription', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<BasicTableDescription item={{imageName:"dsa", model: 'tx'}}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
