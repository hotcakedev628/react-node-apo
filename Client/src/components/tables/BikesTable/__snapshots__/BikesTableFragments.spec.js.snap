// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTableDescription should match snapshot 1`] = `
Array [
  <td>
    <img
        alt="bike"
        src="https://s3.amazonaws.com/bike-images-22/dsa"
        style={
            Object {
                "maxHeight": "60px",
              }
        }
    />
</td>,
  <td>
    tx
</td>,
  <td />,
  <td />,
]
`;
